src/error.rs

```diff
@@ -1,11 +1,19 @@
+
 use defines::AfError;
 use std::error::Error;
+use std::sync::Mutex;
 
 pub type ErrorCallback = Fn(AfError);
 
 pub static mut HANDLE_ERROR: &'static ErrorCallback = &handle_error_general; 
```

Comment:
Can you try a reference to a function (instead of a function) or even a Box maybe?